File: Easy/Delete Middle of Linked List.java
//{ Driver Code Starts
import java.util.*;
import java.io.*;

class Node{
    int data;
    Node next;
    
    Node(int x){
        data = x;
        next = null;
    }
    
}
class GFG{
	static void printList(Node node) 
	{ 
		while (node != null) 
		{ 
			System.out.print(node.data + " "); 
			node = node.next; 
		} 
		System.out.println(); 
	}
    public static void main(String args[]) throws IOException { 
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t > 0){
        	int n = sc.nextInt();
            Node head = new Node(sc.nextInt());
            Node tail = head;
            for(int i=0; i<n-1; i++)
            {
                tail.next = new Node(sc.nextInt());
                tail = tail.next;
            }
            Solution g = new Solution();
            head = g.deleteMid(head);
            printList(head); 
            t--;
        }
    } 
} 
   

// } Driver Code Ends


/*
class Node
{
    int data;
    Node next;
    Node(int d)  { data = d;  next = null; }
}
*/

class Solution
{
    Node deleteMid(Node head)
    {
        // This is method only submission.
        // You only need to complete the method.
        
        Node copia = head;
        Node copia2 = head;
        
        int n = 0;
        
        while (copia != null)
        {
            n++;
            copia = copia.next;
        }
        
        if (n<2) return head;
        
        int mid = n/2;
        if (n%2 == 0) n++;
        
        while (mid>1)
        {
            copia2 = copia2.next;
            mid--;
        }
        
        copia2.next = copia2.next.next;
        
        
        return head;    
    }
}

File: Easy/Delete Middle of Linked List.cpp
//{ Driver Code Starts
#include <bits/stdc++.h> 
using namespace std; 
struct Node
{
    int data;
    struct Node* next;
    
    Node(int x){
        data = x;
        next = NULL;
    }
};



// } Driver Code Ends
/* Link list Node:

struct Node
{
    int data;
    struct Node* next;

    Node(int x)
    {
        data = x;
        next = NULL;
    }
};

*/

// Deletes middle of linked list and returns head of the modified list
class Solution
{
    public:
    Node* deleteMid(Node *head)
    {
        // Your Code Here
        int n = 0;
        Node *copy = head;
        
        while (copy)
        {
            n++;
            copy = copy->next;
        }
        
        int mid = n / 2;
        
        //cout << "n: " << n << " mid: " << mid << endl;
        
        Node *copy2 = head;
        
        for (int i = 0; i < mid - 1; i++)
        {
            copy2 = copy2->next;
        }
        
        //cout << copy2->data << endl;
        
        if (!copy2->next)
        {
            return NULL;
        }
        
        if (copy2->next && !copy2->next->next)
        {
            copy2->next = NULL;
        }
        else if (copy2->next && copy2->next->next)
        {
            copy2->next = copy2->next->next;
        }

        return head;
    }
};

//{ Driver Code Starts.



void printList(Node* node) 
{ 
	while (node != NULL) { 
		cout << node->data <<" "; 
		node = node->next; 
	}  
	cout<<"\n";
} 
int main() 
{ 
	int t;
	cin>>t;
	while(t--)
	{
		int n;
		cin>>n;

		int data;
		cin>>data;
		struct Node *head = new Node(data);
		struct Node *tail = head;
		for (int i = 0; i < n-1; ++i)
		{
			cin>>data;
			tail->next = new Node(data);
			tail = tail->next;
		}
		Solution obj;
		head = obj.deleteMid(head);
		printList(head); 
	}
	return 0; 
} 



// } Driver Code Ends